src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -77,31 +87,146 @@ def increment_project_duration_counter(
+        """
+        Returns a sorted list of bucketed timestamps paired with a histogram-like dictionary of
+        symbolication durations made during some timestamp for some given project.
+
+        For a given `{duration:count}` entry in the dictionary bound to a specific `timestamp`:
+
+        - `duration` represents the amount of time it took for a symbolication request to complete.
+        Durations are bucketed by 10secs, meaning that a `duration` of `30` covers all requests that
+        took between 30-39 seconds.
+
+        - `count` is the number of symbolication requests that took some amount of time within the
+        range of `[duration, duration+10)` to complete.
+
+        This may throw an exception if there is some sort of issue fetching durations from the redis
+        store.
+        """
+        key_prefix = f"{self.histogram_key_prefix()}:{project_id}:"
+        keys = sorted(
+            self.cluster.scan_iter(
+                match=key_prefix + "*",
+            )
+        )
+
+        for key in keys:
+            _, timestamp_raw = key.split(key_prefix)
+            try:
+                timestamp_bucket = int(timestamp_raw)
+            except Exception:
+                logger.warning(
+                    'Ignoring symbolicator metric: project %s key %s, timestamp could not be converted to int: "%s"',
+                    project_id,
+                    key,
+                    timestamp_raw,
+                )
+                continue
+
+            histogram_raw = self.cluster.hgetall(key)
+            histogram = base.BucketedDurations({})
+            try:
+                for (duration_raw, count_raw) in histogram_raw.items():
+                    duration = int(duration_raw)
+                    count = int(count_raw)
+                    histogram[duration] = count
+            except Exception:
+                logger.warning(
+                    "Ignoring symbolicator metric: project %s key %s, unable to parse histogram",
+                    project_id,
+                    key,
+                    extra={"histogram": histogram_raw},
+                )
+                continue
+            yield base.DurationHistogram(timestamp=timestamp_bucket, histogram=histogram)
+
     def get_lpq_projects(self) -> Set[int]:
         """
         Fetches the list of projects that are currently using the low priority queue.
 
         Returns a list of project IDs.
+
+        This may throw an exception if there is some sort of issue fetching the list from the redis
+        store.
         """
         return set(
             filter(
                 None,
-                {
-                    _to_int(project_id_raw)
-                    for project_id_raw in self.cluster.smembers(LPQ_MEMBERS_KEY)
-                },
+                {int(project_id_raw) for project_id_raw in self.cluster.smembers(LPQ_MEMBERS_KEY)},
```

Comment:
so i've gone and updated https://github.com/getsentry/sentry/pull/28757 accordingly since that removes `_to_int` anyways.

as you've noticed, usages of this perform Set operations on its return value so it may not be worth returning an iterable. an iterable isn't capable of guaranteeing is that all of the elements are unique, which is useful for places that use `get_lpq_projects`.